### PR TITLE
refactor: use zerocopy for `IndexHeader` instead of manually slicing

### PR DESCRIPTION
**Pull request by aatifsyed**

### CODE REVIEW

**Review by lemmih — APPROVED** [review]

**Review by hanabi1224 — APPROVED** [review]

File: .config/forest.dic [hunk 1]
```diff
@@ -27,6 +27,7 @@ crypto
 daemon
 daemonize
 DB/S
+deserialized
```
Comment (LesnyRumcajs) [inline .config/forest.dic:30]: we won't have to add `deserialize` next time (I think [it](https://github.com/drahnr/cargo-spellcheck/blob/dff48db8ca954fce14a0cd5aea127ce59a929624/hunspell-data/en_US.aff#L67-L71) should work)
```suggestion
deserialize/D
```

File: src/db/car/forest.rs [hunk 2]
```diff
@@ -116,9 +116,8 @@ impl<ReaderT: super::RandomAccessFileReader> ForestCar<ReaderT> {
         let mut footer_buffer = [0; ForestCarFooter::SIZE];
         cursor.read_exact(&mut footer_buffer)?;
 
-        let footer = ForestCarFooter::try_from_le_bytes(footer_buffer).ok_or(io::Error::new(
-            io::ErrorKind::InvalidData,
-            "Data not recognized as ForestCAR.zst",
+        let footer = ForestCarFooter::try_from_le_bytes(footer_buffer).ok_or(invalid_data(
+            format!("not recognizable as a `{}` file", FOREST_CAR_FILE_EXTENSION),
```
Comment (LesnyRumcajs) [inline src/db/car/forest.rs:120]:
```suggestion
            format!("not recognizable as a `{FOREST_CAR_FILE_EXTENSION}` file"),
```

**Review by LesnyRumcajs — APPROVED** [review]

**Comment by LesnyRumcajs** [comment]
@aatifsyed In the future, please use the PR template with checkboxes and stuff.

File: src/db/car/forest.rs (hunk 2 above)
Comment (aatifsyed) [reply src/db/car/forest.rs:120]: I prefer keeping out-of-line for now - it has better integration with `rust-analyzer`

**Review by hanabi1224 — APPROVED** [review]

**Pull request merged**